ibis/expr/table_method.py

```diff
@@ -0,0 +1,75 @@
+import functools
+import inspect
+
+
+class bound_table_method:
+    """A bound method on a table."""
+
+    __slots__ = ("_func", "_obj")
+
+    def __init__(self, func, obj):
+        self._func = func
+        self._obj = obj
+
+    @property
+    def __doc__(self):
+        return self._func.__doc__
+
+    @property
+    def __name__(self):
+        return self._func.__name__
+
+    @property
+    def __signature__(self):
```

Comment:
Why is this required?

Reply:
For signature inference of the method. `__signature__` is used by `inspect.signature` and other introspection tools (this mainly comes up when working in `ipython`/`jupyter`).